panel_segmentation/utils.py

```diff
@@ -0,0 +1,709 @@
+    box : list
+        A list of float pixel values containing the coordinates of a bounding
+        box in the format [xmin, ymin, xmax, ymax].
+    img_center_lat : float
+        Latitude coordinate of the image center.
+    img_center_lon : float
+        Longitude coordinate of the image center.
+    image_x_pixels : int
+        The x width of the image in pixels.
+    image_y_pixels : int
+        The y height of the image in pixels.
+    zoom_level : int
+        Zoom level of the image.
+
+    Returns
+    -------
+    (box_lat, box_lon) : tuple
+        The (latitude, longitude) coordinates of the centroid of a box.
+    """
+    # Ensure that the inputs are of the correct type
+    if not isinstance(box, list):
+        raise TypeError("box variable must be of type list.")
+    if not isinstance(img_center_lat, float):
+        raise TypeError("img_center_lat variable must be of type float.")
+    if not isinstance(img_center_lon, float):
+        raise TypeError("img_center_lon variable must be of type float.")
+    if not isinstance(image_x_pixels, int) or isinstance(image_x_pixels, bool):
+        raise TypeError("image_x_pixels variable must be of type int.")
+    if not isinstance(image_y_pixels, int) or isinstance(image_y_pixels, bool):
+        raise TypeError("image_y_pixels variable must be of type int.")
+    if not isinstance(zoom_level, int) or isinstance(zoom_level, bool):
+        raise TypeError("zoom_level variable must be of type int.")
+    image_center_pixels_x, image_center_pixels_y = (image_x_pixels/2,
+                                                    image_y_pixels/2)
+    xmin, ymin, xmax, ymax = (float(box[0]), float(box[1]),
+                              float(box[2]), float(box[3]))
+    cx = int((xmin + xmax) / 2)
+    cy = int((ymin + ymax) / 2)
+    # Get the difference in meters between the main centroid and
+    # label centroid, based on the image zoom level
+    meter_pixel_conversion = meter_pixel_zoom_dict[zoom_level]
+    lon_translation_meters = ((cx - image_center_pixels_x) *
+                              meter_pixel_conversion)
+    lat_translation_meters = ((image_center_pixels_y - cy) *
+                              meter_pixel_conversion)
+    box_lat, box_lon = translateLatLongCoordinates(
+        latitude=img_center_lat,
+        longitude=img_center_lon,
+        lat_translation_meters=lat_translation_meters,
+        long_translation_meters=lon_translation_meters)
+    return (box_lat, box_lon)
+
+
+def binaryMaskToPolygon(mask):
+    """
+    Convert a binary mask output (from a deep learning model) to a list of
+    polygon coordinates, which can later be converted to latitude-longitude
+    coordinates.
+
+    Parameters
+    -----------
+    mask : nparray
+        A binary mask output from a deep learning model, which can be converted
+        to a polygon.
+
+    Returns
+    -------
+    contours_new : list
+        A list of (x,y) coordinates for a polygon.
```

Comment:
We should say this by image pixel location (so ppl know it's not returning by lat-lon)